src/event.cpp

```diff
@@ -488,6 +489,11 @@ void event_print(io_streams_t &streams, const wcstring &type_filter) {
     }
 }
 
+void event_fire_generic(parser_t &parser, wcstring name, const wcharz_t argv[], int argc) {
```

Comment:
Note: *Never* use array function arguments in C++. They don't really *exist*.

C++ arrays fall apart and decay into pointers at the first convenience, and they don't actually carry around anything useful (like, you know, length?).

This should be `const wcharz_t *argv`.

Anyway, the construction guff can be sidestepped by just going the long way around:

```c++
void event_fire_generic(parser_t &parser, wcstring name, const wcharz_t *argv, int argc) {
    wcstring_list_t args_vec{};
    for (int i = 0; i < argc; i++) {
        args_vec.push_back(argv[i]);
    }
    event_fire_generic(parser, std::move(name), std::move(args_vec));
}
```

I *think* the kerfuffle could be triggered by const-issues, but tbh I'm so tired of C++ I don't want to learn user conversion operators *now*.

Reply:
> Note: Never use array function arguments in C++. They don't really exist.
>
> C++ arrays fall apart and decay into pointers at the first convenience, and they don't actually carry around anything useful (like, you know, length?).

oh yeah, I know that from C, I just saw it as an annotation for humans that `argv` isn't a single `wcharz_t`, but rather a pointer to several of them. Not really necessary given the argument name though.

> Anyway, the construction guff can be sidestepped by just going the long way around

Applied, thanks!